action.yml

```diff
@@ -1,60 +1,188 @@
+    description: "The tutor version matrix to use."
     required: true
   shell_file_to_run:
-    description: 'The path of the shell file to run the integration tests.'
+    description: "The path of the shell file to run the integration tests."
+    required: true
+  openedx_extra_pip_requirements:
+    description: "Optional extra pip requirements to install in Open edX. E.g: 'package1==1.0 package2>=2.0'"
+    required: false
+    default: ""
+  fixtures_file:
+    description: "Optional path to the plugin's fixtures file to load."
+    required: false
+  openedx_imports_test_file_path:
+    description: "Path to the file that contains the test function for validating Open edX imports. This should be a Python file within your project."
+    required: false
+  openedx_imports_test_function_name:
+    description: "Name of the function in the specified file that executes the import tests for Open edX."
     required: false
 
 runs:
-  using: 'composite'
+  using: "composite"
   steps:
-  - name: Create a Virtual Environment
-    run: |
-      pip install virtualenv
-      virtualenv venv
-    shell: bash
-
-  - name: Install Tutor
-    run: |
-      source venv/bin/activate
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        git clone --branch=nightly --depth=1 https://github.com/overhangio/tutor.git
-        chmod 777 tutor -R
-        pip install -e "./tutor[full]"
-      else
-        pip install "tutor$INPUT_TUTOR_VERSION"
-      fi
-    shell: bash
-    env:
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Configure Tutor and Launch
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor --version
-      TUTOR_ROOT="$(pwd)" tutor config save
-      TUTOR_ROOT="$(pwd)" tutor mounts add lms,cms,lms-worker,cms-worker:$(pwd)/$INPUT_APP:/openedx/$INPUT_APP
-      chmod 777 . -R
-      if [ "$INPUT_TUTOR_VERSION" = "nightly" ]; then
-        TUTOR_ROOT="$(pwd)" tutor images build openedx
-      fi
-      TUTOR_ROOT="$(pwd)" tutor local launch -I
-    shell: bash
-    env:
-      INPUT_APP: ${{ inputs.app_name }}
-      INPUT_TUTOR_VERSION: ${{ inputs.tutor_version }}
-
-  - name: Run integration tests in lms
-    if: ${{ inputs.shell_file_to_run }}
-    run: |
-      source venv/bin/activate
-      TUTOR_ROOT="$(pwd)" tutor local run lms bash $INPUT_SHELL_FILE
-    shell: bash
-    env:
-      INPUT_SHELL_FILE: /openedx/${{ inputs.app_name }}/${{ inputs.shell_file_to_run }}
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        path: ${{ inputs.app_name }}
+
+    - name: Adjust permissions to execute Tutor commands
+      run: |
+        chmod 777 . -R
+      shell: bash
+
+    - name: Set Tutor environment variables
+      run: |
+        echo "LMS_HOST=local.edly.io" >> "$GITHUB_ENV"
+        echo "CMS_HOST=studio.local.edly.io" >> "$GITHUB_ENV"
+        echo "TUTOR_ROOT=$(pwd)" >> "$GITHUB_ENV"
+        echo "TUTOR_PLUGINS_ROOT=$(pwd)/plugins/" >> "$GITHUB_ENV"
```

Comment:
Does [this](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#example-of-a-multiline-string) work?

```
      {
        LMS_HOST=local.edly.io
        CMS_HOST=studio.local.edly.io
        TUTOR_ROOT=$(pwd)
        TUTOR_PLUGINS_ROOT=$(pwd)/plugins/
        echo EOF
      } >> "$GITHUB_ENV"
```

Reply:
Did not worked exactly like this, but It worked using [Heredoc](https://github.com/eduNEXT/integration-test-in-tutor/pull/4/commits/839bffaf3eedf2bf3364188f848e7a0ab26e9e53). Thanks!

Reply:
That exact code didn't work, but using [heredoc](https://github.com/eduNEXT/integration-test-in-tutor/pull/4/commits/839bffaf3eedf2bf3364188f848e7a0ab26e9e53) did, which still helped me avoid repetition and made the code more readable. Thanks!